Description: Use _Static_assert
 The previous hack for making compile-time assertions no longer works
 with GCC 6.  Use the built-in _Static_assert instead.
Origin: other, package ucl (1.03+repack-4) patch 04-Static-assert.patch
Bug-Debian: https://bugs.debian.org/812054
Forwarded: no
Last-Update: 2016-07-05

Index: b/src/miniacc.h
===================================================================
--- a/src/miniacc.h
+++ b/src/miniacc.h
@@ -1699,6 +1699,9 @@ extern "C" {
 #  endif
 #endif
 #if !defined(ACC_COMPILE_TIME_ASSERT_HEADER)
+# define ACC_COMPILE_TIME_ASSERT_HEADER(e) _Static_assert(e, #e);
+#endif
+#if !defined(ACC_COMPILE_TIME_ASSERT_HEADER)
 #  if (ACC_CC_AZTECC || ACC_CC_ZORTECHC)
 #    define ACC_COMPILE_TIME_ASSERT_HEADER(e)  extern int __acc_cta[1-!(e)];
 #  elif (ACC_CC_DMC || ACC_CC_SYMANTECC)
@@ -1710,6 +1713,9 @@ extern "C" {
 #  endif
 #endif
 #if !defined(ACC_COMPILE_TIME_ASSERT)
+# define ACC_COMPILE_TIME_ASSERT(e) _Static_assert(e, #e);
+#endif
+#if !defined(ACC_COMPILE_TIME_ASSERT)
 #  if (ACC_CC_AZTECC)
 #    define ACC_COMPILE_TIME_ASSERT(e)  {typedef int __acc_cta_t[1-!(e)];}
 #  elif (ACC_CC_DMC || ACC_CC_PACIFICC || ACC_CC_SYMANTECC || ACC_CC_ZORTECHC)
